src/index.ts

```diff
@@ -36,33 +34,73 @@ const createWindow = (): void => {
   });
 };
 
-// This method will be called when Electron has finished
-// initialization and is ready to create browser windows.
-// Some APIs can only be used after this event occurs.
-// React Developer Tools are installed before the window is created.
-app.on("ready", async () => {
-  const reactDevToolsPath = path.join(
-    __dirname,
-    "..",
-    "..",
-    "src",
-    "assets",
-    "tools",
-    "reactDevTools",
-  );
-  try {
-    await session.defaultSession.loadExtension(reactDevToolsPath);
-    console.log("Added Extension: React Developer Tools");
-    createWindow();
-  } catch (err) {
-    console.log("An error occurred: ", err);
-    createWindow();
-  }
+app.whenReady().then(() => {
+  protocol.registerHttpProtocol("intracom", (request, callback) => {
+    const url = request.url.substring(11);
+    const filePath = path.normalize(`${__dirname}/${url}`);
+    callback({ path: filePath });
+  });
+  // This method will be called when Electron has finished
+  // initialization and is ready to create browser windows.
+  // Some APIs can only be used after this event occurs.
+  // React Developer Tools are installed before the window is created.
+  app.on("ready", async () => {
```

Comment:
`app.whenReady` is the same as `app.on('ready'`, you should be able to remove the `app.on('ready', ()=> {});` wrapper on the following code:
```
const reactDevToolsPath = path.join(
      __dirname,
      "..",
      "..",
      "src",
      "assets",
      "tools",
      "reactDevTools",
    );
    try {
      await session.defaultSession.loadExtension(reactDevToolsPath);
      console.log("Added Extension: React Developer Tools");
      createWindow();
    } catch (err) {
      console.log("An error occurred: ", err);
      createWindow();
    }
```
and just put the code directly in `app.whenReady`.